File: src/main/java/nl/hand/made/vending/machine/repository/ProductRepository.java
package nl.hand.made.vending.machine.repository;

import nl.hand.made.vending.machine.repository.model.ProductEntity;
import nl.hand.made.vending.machine.repository.model.UserEntity;
import org.springframework.data.repository.CrudRepository;

import javax.transaction.Transactional;
import java.util.List;
import java.util.Optional;

@Transactional
public interface ProductRepository extends CrudRepository<ProductEntity, Integer> {

    List<ProductEntity> findAll();

    List<ProductEntity> findAllByAmountAvailableIsGreaterThan(Integer amount);

    Optional<ProductEntity> findByIdAndSeller(Integer id, UserEntity userEntity);
}

File: src/main/kotlin/nl/hand/made/vending/machine/repository/ProductRepository.kt
package nl.hand.made.vending.machine.repository

import nl.hand.made.vending.machine.repository.model.ProductEntity
import nl.hand.made.vending.machine.repository.model.UserEntity
import org.springframework.data.repository.CrudRepository
import javax.transaction.Transactional

@Transactional
interface ProductRepository : CrudRepository<ProductEntity, Int> {
    override fun findAll(): List<ProductEntity>
    fun findAllByAmountAvailableIsGreaterThan(amount: Int): List<ProductEntity>
    fun findByIdAndSeller(id: Int, userEntity: UserEntity): ProductEntity?
}